src/agent/border_agent.cpp

```diff
@@ -229,10 +233,16 @@ void BorderAgent::PublishService(void)
 {
     assert(mNetworkName[0] != '\0');
     assert(mExtPanIdInitialized);
+    assert(mExtAddrInitialized);
     assert(mThreadVersion != 0);
 
     const char *             versionString = ThreadVersionToString(mThreadVersion);
-    Mdns::Publisher::TxtList txtList{{"nn", mNetworkName}, {"xp", mExtPanId, sizeof(mExtPanId)}, {"tv", versionString}};
+    Mdns::Publisher::TxtList txtList{{"nn", mNetworkName},
+                                     {"xp", mExtPanId, sizeof(mExtPanId)},
+                                     {"tv", versionString},
+                                     // "dd" represents for device discriminator which
+                                     // should always be the extended address.
+                                     {"dd", mExtAddr, sizeof(mExtAddr)}};
```

Comment:
Should we re-publish the service when the extended address changes (e.g. after commission)

Reply:
I think the discriminator should never be changed even if the Border Agent has been commissioned into another Thread network.

Reply:
> I think the discriminator should never be changed even if the Border Agent has been commissioned into another Thread network.

Indeed. The extended address should not change after commissioning. However, currently we don't re-publish the service when the device reattaches (after which the extended address might change). I think we need to re-publish the mDNS service when the Thread state is changed from detached to attached (preferably in a separate PR).

Reply:
@simonlingoogle Actually, we are advertising the EUI64 of the border agent device.
```c++
case kEventEui64:
    {
        otExtAddress eui64;

        otLinkGetFactoryAssignedIeeeEui64(mInstance, &eui64);
        EventEmitter::Emit(kEventEui64, eui64.m8);
        break;
...
```

I changed the name from `mExtAddr` to `mEui64` to avoid confusion.

Reply:
I think we should use the Extended Address. A concern around static identifiers like IEEE EUI-64 is ability to track devices.

Reply:
I gave this some thought yesterday as well, and agree to extended address (for whatever it's worth).

Reply:
> A concern around static identifiers like IEEE EUI-64 is ability to track devices.

Makes sense to me. Will change accordingly.

Reply:
Revised to use the Extended Address, PTAL.